managed/services/agents/agentversion.go

```diff
@@ -23,23 +23,34 @@ import (
 	"github.com/percona/pmm/managed/models"
 )
 
-// PMMAgentSupported checks if pmm agent version satisfies required min version.
-func PMMAgentSupported(q *reform.Querier, pmmAgentID, functionalityPrefix string, pmmMinVersion *version.Version) error {
+// ErrIncompatibleAgentVersion is returned when the version of the running pmm-agent does not support a given functionality.
+var ErrIncompatibleAgentVersion = errors.New("incompatible agent version")
+
+// PMMAgentSupportedByAgentID wraps PMMAgentSupported and accepts agent ID instead of agent model.
+func PMMAgentSupportedByAgentID(q *reform.Querier, pmmAgentID, functionalityPrefix string, pmmMinVersion *version.Version) error {
 	pmmAgent, err := models.FindAgentByID(q, pmmAgentID)
 	if err != nil {
 		return errors.Errorf("failed to get PMM Agent: %s", err)
 	}
-	if pmmAgent.Version == nil {
-		return errors.Errorf("pmm agent %q has no version info", pmmAgentID)
+	return PMMAgentSupported(pmmAgent, functionalityPrefix, pmmMinVersion)
+}
+
+// PMMAgentSupported checks if pmm agent version satisfies required min version.
+func PMMAgentSupported(agentModel *models.Agent, functionalityPrefix string, pmmMinVersion *version.Version) error {
```

Comment:
`PMMAgentSupported` seems more readable. I think in this case, it's okay to also let`PMMAgentSupported` make the db call to find the agent by ID and do the version check.

Now, we have two exported similarly named exported methods.

Reply:
I agree that we don't need 2 funcs, so I've hid one of them. But I prefere to have logic separated from external call (e.g. db calls) for easy testing